packages/govern-console/src/components/Modal/NewActionModal.tsx

```diff
@@ -117,28 +127,18 @@ export const NewActionModal: React.FC<NewActionModalProps> = ({
 
   const onInputABI = (val: string) => {
     if (isJsonString(val)) {
-      abi.current = JSON.parse(val);
+      setAbi(JSON.parse(val));
     } else return;
   };
   const onInputContractAddress = (val: any) => {
-    contractAddress.current = val;
+    setContractAddress(val);
   };
-  const onGenerateClick = () => {
-    if (abi.current === null) return;
-    if (contractAddress.current === null) return;
 
-    console.log(abi.current, contractAddress.current);
-    onGenerate(contractAddress.current, abi.current);
+  const onGenerateClick = () => {
+    if (abi === null) return;
```

Comment:
@Rekard0 why 2 if statements ? 1 if with `||` should be enough and there's no need to check it with `null` . 

```
if(!abi || !contractAddress) return
```